classes/class-p4-custom-taxonomy.php

```diff
@@ -164,42 +169,79 @@ public function register_taxonomy() {
 				'labels'       => $p4_page_type,
 				'show_ui'      => true,
 				'query_var'    => true,
-				'rewrite'      => [ 'slug' => 'page_type' ],
 				'meta_box_cb'  => [ $this, 'create_taxonomy_metabox_markup' ],
 			];
 
 			register_taxonomy( self::TAXONOMY, [ 'p4_page_type', 'post' ], $args );
+
+			// Add a rewrite rule on top of the chain for each slug
+			// of this taxonomy type (e.g.: "publication", "story", etc.)
+			$terms_slugs = $this->get_terms_slugs();
```

Comment:
I think we will need to encapsulate this in a function.
This section have to be executed (maybe with some modifications) when a new term is deleted/created/updated like this.
Also will need to flush the rewrite rules after each change in the current rules

https://github.com/greenpeace/planet4-master-theme/pull/657/files#diff-8f787c8ac52bf86ac5547655d492a4dcR26

Reply:
Good point, will do!